### PR TITLE
Common Resource CFT to TF conversion

### PR DESCRIPTION
**Pull request by mschuchard**

### CODE REVIEW

File: aws-observability-terraform/terraform-aws-observability-common/alb_source.tf [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  content_type  = "AwsElbBucket"
+  name          = var.alb_logs_source_name
+  paused        = false
+  scan_interval = var.scan_interval
+
+  authentication {
+    type     = "AWSRoleBasedAuthentication"
+    role_arn = aws_iam_role.sumologic_source["this"].id
+  }
+
```
Comment (SumoSourabh) [inline aws-observability-terraform/terraform-aws-observability-common/alb_source.tf:15]: We will have to provide cutoff_relative_time=-1d and fields as per CF here, link https://github.com/hashicorp/terraform-provider-sumologic/tree/master/website#common-source-properties

File: aws-observability-terraform/terraform-aws-observability-common/alb_source.tf [hunk 2]
```diff
@@ -0,0 +1,53 @@
+resource "sumologic_elb_source" "this" {
+  for_each = toset(var.manage_alb_logs_source ? ["this"] : [])
+
+  category      = "aws/observability/alb/logs"
```
Comment (SumoSourabh) [inline aws-observability-terraform/terraform-aws-observability-common/alb_source.tf:4]: Can we make this as a variable for all sources

**Comment by mschuchard** [comment]
Above commit represents prior to merging RCE into common.

**Comment by mschuchard** [comment]
@SumoSourabh @arunpatyal Above commit represents migrating root cause explorer into common. My testing had a successful outcome, but please review for anything I may have missed.

File: terraform-aws-observability-common/alb_source.tf [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+  authentication {
+    type     = "AWSRoleBasedAuthentication"
+    role_arn = aws_iam_role.sumologic_source["this"].arn
+  }
+
+  path {
+    type            = "S3BucketPathExpression"
+    bucket_name     = var.manage_alb_s3_bucket ? aws_s3_bucket.common["this"].id : var.alb_s3_bucket
+    path_expression = "*AWSLogs/*/elasticloadbalancing/${var.alb_s3_bucket_path_expression}"
```
Comment (SumoSourabh) [inline terraform-aws-observability-common/alb_source.tf:26]: Can you please check the path expression. It should be based on condition. If bucket is created than we add a default path expression. If customer provided bucket is used, then pass the customer provided path expression

Comment (SumoSourabh) [reply terraform-aws-observability-common/alb_source.tf:26]: for both alb and CloudTrail

**Pull request merged**